### PR TITLE
Withdraw flow and lender stats card v2

### PR DESCRIPTION
**Pull request by mliu**
image

### CODE REVIEW

File: packages/huma-shared/src/v2/utils/poolContract.ts [hunk 1]
```diff
@@ -146,3 +146,21 @@ export const getTrancheVaultAssetsV2 = async (
 
   return trancheVaultContract.totalAssets()
 }
+
+export const getTrancheToSharesV2 = async (
```
Comment (shan-57blocks) [inline packages/huma-shared/src/v2/utils/poolContract.ts:150]: I feel like 'getTrancheSharesV2' is a better name?

Comment (shan-57blocks) [inline packages/huma-shared/src/v2/utils/poolContract.ts:150]: Can we add the unit tests?

File: packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    setCurrentAmount(newAmount)
+    dispatch(setWithdrawAmount(Number(newAmount)))
+  }
+
+  const handleAction = () => {
+    dispatch(setStep(WIDGET_STEP.Transfer))
+  }
+
+  return (
+    <WrapperModal title='Withdraw' subTitle='Choose withdraw type'>
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx:79]: From what Bin told me, now user cannot choose withdraw type.

File: packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  poolUnderlyingToken: UnderlyingTokenInfo
+}
+
+export function ChooseAmount({
+  poolUnderlyingToken,
+}: Props): React.ReactElement | null {
+  const theme = useTheme()
+  const dispatch = useAppDispatch()
+  const { symbol } = poolUnderlyingToken
+  const [currentAmount, setCurrentAmount] = useState<number | string>(0)
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx:30]: I think we need to show what's the correspondent shares for the amount

File: packages/huma-widget/src/components/Lend/withdrawV2/4-Done.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+export function Done({
+  poolUnderlyingToken,
+  handleAction,
+}: Props): React.ReactElement {
+  const { symbol } = poolUnderlyingToken
+  const { withdrawAmount } = useAppSelector(selectWidgetState)
+
+  const content = [
+    `You successfully withdrew ${formatMoney(withdrawAmount)} ${symbol}.`,
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Lend/withdrawV2/4-Done.tsx:20]: Actually there will be no withdraw immediately, just the request. Need to rephrase it.

**Review by shan-57blocks — APPROVED** [review]

File: packages/huma-widget/src/components/Lend/withdrawV2/index.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  )
+
+  if (!poolInfo || !poolUnderlyingToken) {
+    return null
+  }
+
+  return (
+    <WidgetWrapper
+      isOpen
+      loadingTitle='Withdraw'
```
Comment (shan-57blocks) [inline packages/huma-widget/src/components/Lend/withdrawV2/index.tsx:92]: After confirmed with Erbil, here should be 'Withdraw USDC'

File: packages/huma-shared/src/v2/utils/poolContract.ts (hunk 1 above)
Comment (mliu) [reply packages/huma-shared/src/v2/utils/poolContract.ts:150]: done

File: packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx (hunk 3 above)
Comment (mliu) [reply packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx:30]: I don't think shares have any meaning to the user. The asset amount is what matters here.

File: packages/huma-widget/src/components/Lend/withdrawV2/4-Done.tsx (hunk 4 above)
Comment (mliu) [reply packages/huma-widget/src/components/Lend/withdrawV2/4-Done.tsx:20]: Yep, I've reached out to Du and Erbil on Figma to also leave this feedback.

File: packages/huma-widget/src/components/Lend/withdrawV2/index.tsx (hunk 5 above)
Comment (mliu) [reply packages/huma-widget/src/components/Lend/withdrawV2/index.tsx:92]: done

File: packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx (hunk 2 above)
Comment (mliu) [reply packages/huma-widget/src/components/Lend/withdrawV2/2-ChooseAmount.tsx:79]: will remove

File: packages/huma-shared/src/v2/utils/poolContract.ts (hunk 1 above)
Comment (mliu) [reply packages/huma-shared/src/v2/utils/poolContract.ts:150]: added

**Pull request merged**